tapchannel/aux_invoice_manager.go

```diff
@@ -121,9 +153,25 @@ func (s *AuxInvoiceManager) handleInvoiceAccept(_ context.Context,
 
 	// No custom record on the HTLC, so we have nothing to do.
 	if len(req.WireCustomRecords) == 0 {
+		// If there's no wire custom records and the invoice is an asset
+		// invoice do not settle the invoice. Since we are asking for
+		// assets in the invoice, we may not let this HTLC go through
+		// as it is not carrying assets. This could lead to undesired
+		// behavior where the asset invoice may be settled by accepting
+		// sats instead of assets.
+		//
+		// TODO(george): Strict-forwarding could be configurable?
+		if isAssetInvoice(req.Invoice, s) {
+			resp.CancelSet = true
+		}
+
 		return resp, nil
 	}
 
+	if req.Invoice.PaymentAddr == nil {
```

Comment:
Why is this needed? I'm not sure if the payment address is always set for AMP payments (or just for MPP)...
If this is because of unit tests, perhaps just checking the invoice for being non-nil?